pandas/io/tests/test_gbq.py

```diff
@@ -517,18 +554,72 @@ def test_should_properly_handle_valid_integers(self):
+                    (select 1 as NULLABLE_INTEGER),
+                    (select null as NULLABLE_INTEGER)'''
+        df = gbq.read_gbq(query, project_id=_get_project_id(),
+                          private_key=_get_private_key_path())
+        tm.assert_frame_equal(
+            df, DataFrame({'NULLABLE_INTEGER': [1, None]}).astype(object))
+
+    def test_should_properly_handle_valid_longs(self):
+        query = 'SELECT 1 << 62 as VALID_LONG'
+        df = gbq.read_gbq(query, project_id=_get_project_id(),
+                          private_key=_get_private_key_path())
+        tm.assert_frame_equal(
+            df, DataFrame({'VALID_LONG': [1 << 62]}))
+
+    def test_should_properly_handle_nullable_longs(self):
+        query = '''select * from
+                    (select 1 << 62 as NULLABLE_LONG),
+                    (select null as NULLABLE_LONG)'''
+        df = gbq.read_gbq(query, project_id=_get_project_id(),
+                          private_key=_get_private_key_path())
+        tm.assert_frame_equal(
+            df, DataFrame({'NULLABLE_LONG': [1 << 62, None]}).astype(object))
+
     def test_should_properly_handle_null_integers(self):
         query = 'SELECT INTEGER(NULL) as NULL_INTEGER'
         df = gbq.read_gbq(query, project_id=_get_project_id(),
                           private_key=_get_private_key_path())
-        tm.assert_frame_equal(df, DataFrame({'NULL_INTEGER': [np.nan]}))
+        tm.assert_frame_equal(df, DataFrame({'NULL_INTEGER': [None]}))
 
     def test_should_properly_handle_valid_floats(self):
+        from math import pi
         query = 'SELECT PI() as VALID_FLOAT'
         df = gbq.read_gbq(query, project_id=_get_project_id(),
                           private_key=_get_private_key_path())
         tm.assert_frame_equal(df, DataFrame(
-            {'VALID_FLOAT': [3.141592653589793]}))
+            {'VALID_FLOAT': [pi]}))
+
+    def test_should_properly_handle_nullable_floats(self):
+        from math import pi
+        query = '''select * from
+                    (select PI() as NULLABLE_FLOAT),
+                    (select null as NULLABLE_FLOAT)'''
+        df = gbq.read_gbq(query, project_id=_get_project_id(),
+                          private_key=_get_private_key_path())
+        tm.assert_frame_equal(
+            df, DataFrame({'NULLABLE_FLOAT': [pi, None]}))
+
+    def test_should_properly_handle_valid_doubles(self):
+        from math import pi
+        query = 'SELECT PI() * POW(10, 307) as VALID_DOUBLE'
+        df = gbq.read_gbq(query, project_id=_get_project_id(),
+                          private_key=_get_private_key_path())
+        tm.assert_frame_equal(df, DataFrame(
+            {'VALID_DOUBLE': [pi * 10 ** 307]}))
+
+    def test_should_properly_handle_nullable_doubles(self):
+        from math import pi
+        query = '''select * from
+                    (select PI() * POW(10, 307) as NULLABLE_DOUBLE),
+                    (select null as NULLABLE_DOUBLE)'''
```

Comment:
SQL reserved words should be upper case for consistency with existing code.  We should either use the following or change existing code if there is a preference to change the SQL formatting.
```
        query = '''SELECT * FROM
                    (SELECT PI() * POW(10, 307) AS Nullable_Double),
                    (SELECT NULL AS Nullable_Double)'''
```